pkg/controller/application/application_controller.go

```diff
@@ -82,8 +86,24 @@ func NewController(
 
 	appInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: c.enqueueApp,
-		UpdateFunc: func(_, new interface{}) {
-			c.enqueueApp(new)
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			oldApp, ok := oldObj.(*shipper.Application)
+			if !ok {
+				runtime.HandleError(fmt.Errorf("not a shipper.Application: %#v", oldObj))
+				return
+			}
+
+			newApp, ok := newObj.(*shipper.Application)
+			if !ok {
+				runtime.HandleError(fmt.Errorf("not a shipper.Application: %#v", newObj))
+				return
+			}
+
+			if oldApp.GetResourceVersion() == newApp.GetResourceVersion() {
+				c.enqueueAppAfterDuration(newObj)
```

Comment:
The name of this method is a bit misleading as it provides no access to the specific timeout. I.e.: calling it like: `c.enqueueAppAfterDuration(newObj, time.Second)` would match with my expectations from a method called like this. The method name should give a signal the caller doesn't really have a control over it. Proposed name: `enqueueAppLater(newObj)`. Reading `enqueueAppAfterDuration` code indicates the delay would be chosen based on the resync period. This raises a question: is it a permanent strategy to choose the juffy based on resync period or it's just something that makes sense by now but it's a subject of future reconsideration? I'm mainly asking because initially I saw this code and wanted to propose a method rename: `enqueueAppOnResync`, which eventually made very little sense to me as a resync mechanism will still bting it to the resync stage and therefore might be dropped completely.